content/blog/2021-11-02-agent.md

```diff
@@ -0,0 +1,104 @@
+
+What I personally love in Prometheus project and one of the many reasons why I joined the team, was the laser focus on the project goals. Prometheus was always about moving boundaries when it comes to providing pragmatic, reliable, cheap, yet invaluable metric monitoring. Its ultra stable and robust APIs, querying language and integration protocols (e.g. Remote Write and [Open Metrics](https://openmetrics.io/)) allowed the CNCF metric ecosystem to grow on those strong foundations. This is why we don't see Kubernetes clusters without Prometheus running there. This is also why we can see community exporters for getting metrics about literally everything e.g. [containers](https://github.com/google/cadvisor), [eBPF](https://github.com/cloudflare/ebpf_exporter), [Minecraft server statistics](https://github.com/sladkoff/minecraft-prometheus-exporter) and even [plants health when gardening](https://megamorf.gitlab.io/2019/07/14/monitoring-plant-health-with-prometheus/). 
+
+The strong focus of the Prometheus community allowed other open source projects to grow (e.g. [Cortex](https://cortexmetrics.io/), [Thanos](https://thanos.io/) and more) to extend Prometheus data model beyond single clusters. Not mentioning cloud vendors adopting Prometheus API and data model (e.g. [Amazon Managed Prometheus](https://aws.amazon.com/prometheus/), [Google Cloud Managed Prometheus](https://cloud.google.com/stackdriver/docs/managed-prometheus), [Grafana Cloud](https://grafana.com/products/cloud/) and more). If you are looking for a single reason why Prometheus project is so successful, it is this: *Focusing the monitoring community on what matters*.
+
+In this blog post I would love to introduce you a new mode of running Prometheus called "agent". It is built directly into the Prometheus binary. The agent mode blocks some of the Prometheus features and optimizes the binary for scraping and remote writing to remote location. While reducing number of features on the solution sounds like a regression, I will explain you why it is a game changer for certain deployments in the CNCF ecosystem. I am personally super excited for this! (:
+
+## History of the Forwarding Use Case
+
+The core design of Prometheus has been unchanged for the whole project lifetime. Inspired by the Google Borgmon, you can deploy it next to the application you want to monitor, tell Prometheus where they are and allow Prometheus to scrape the current values of their metrics every configured interval. Such collection method, which is often referred as "pull model" is the core principle that allows Prometheus to be lightweight and ultra reliable. Furthermore, it allows applications and exporters to be dead simple as they need to provide a simple human-readable HTTP page with current metric values (in Open Metric format), without complex push infrastructure as we typically see. Overall, in simple view, typical Prometheus monitoring deployment looks as presented below:
```

Comment:
Should we refer to Julius' blog post? https://prometheus.io/blog/2016/07/23/pull-does-not-scale-or-does-it/